lib/improver/threshold.py

```diff
@@ -279,7 +286,10 @@ def process(self, input_cube):
             # if upper and lower bounds are equal, set a deterministic 0/1
             # probability based on exceedance of the threshold
             if bounds[0] == bounds[1]:
-                truth_value = cube.data > threshold
+                if self.equal_thresh_ok != self.below_thresh_ok:
+                    truth_value = cube.data >= threshold
+                else:
+                    truth_value = cube.data > threshold
```

Comment:
We probably also need to consider the implications for the `spp__relative_to_threshold` attribute, which we use in a number of places, as currently this is either "above" or "below". We need to introduce "above_or_equal_to" and possibly "below_or_equal_to" to better represent what we've calculated.